### PR TITLE
Headless webclipper to convert notes with only URLs body shared from mobile apps into clipped simplied page content

### PR DESCRIPTION
**Pull request by joybinchen**
I have implemented a headless clipper server in HeadlessClient directory.

Most files in this directory are just build scripts, the actual logic will copy from ReactNativeCient/lib, CliClient and Clipper/joplin-clipper/content_scripts in build time. I think I have made these codes compatible with those three clients.

This service can help joplin mobile app to convert URLs shared from other apps into clipped simplied page.

When this headless client sync from various sync target periosely, it watches created and updated notes with only URL in body, and converts such notes into clipped simplied page, just like what joplin-clipper + ClipperServer is doing on browser webExtention and ElectronClient.

This service will perform best when it's running on a NAS (maybe from a docker container with xvfb) and use local filesystem as the sync.target. Note sync action will be triggered instantly when the files in the target directory get changed. After web page loading and clipping finished in a few second, updated note body in markdown format will be sync back to sync target in a second.

### CODE REVIEW

**Comment by joybinchen** [comment]
URL note body to clipSimplifiedPage auto conversion for ElectronClient is implemented, too. Will be commited later.

File: readme/headless-clipper.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
+HeadlessClient contains a headless clipper server.
+
+This service can help joplin mobile app to convert URLs shared from other apps into clipped simplied page.
+
+When this headless client sync from various sync target periosely, it watches created and updated notes with only URL in body, and converts such notes into clipped simplied page, just like what joplin-clipper + ClipperServer is doing on browser webExtention and ElectronClient.
```
Comment (foxmask) [inline readme/headless-clipper.md]: periosely ? is there a typo ?

Comment (tessus) [reply readme/headless-clipper.md]: I think 
```suggestion
periodically
```

was meant. In any case, the entire paragraph could need a rewrite, but I understood what the author meant to say, so these changes are just cosmetically.

**Comment by laurent22** [comment]
Thanks for the pull request. It's quite a major feature so I'll need a bit of time to review it.

**Comment by laurent22** [comment]
This is quite difficult to merge due to the amount of code and complexity. It's almost like a project on its own, and I won't be able to maintain it at the moment.

I would suggest to create a new repository for this, and I'd be glad to link to it from the Readme and website. Once this sub-project becomes more mature and used by more people, maybe we could start looking at how it could be better integrated to Joplin and make it part of the main repository. What do you think?

**Comment by laurent22** [comment]
At the moment this cannot be merged, but as mentioned above if you can somehow make it a separate project, I'd be glad to link to it and maybe later to integrate it to the main app.

**Pull request closed without merging**